src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/config/AuditConfig.java

```diff
@@ -0,0 +1,195 @@
+package com.amazon.opendistroforelasticsearch.security.auditlog.config;
+
+import com.amazon.opendistroforelasticsearch.security.auditlog.impl.AuditCategory;
+import com.amazon.opendistroforelasticsearch.security.support.ConfigConstants;
+import org.elasticsearch.common.settings.Settings;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.List;
+
+public class AuditConfig {
+    private static final List<String> DEFAULT_IGNORED_USERS = Collections.singletonList("kibanaserver");
+    private static final List<String> DEFAULT_DISABLED_CATEGORIES =
+            Arrays.asList(AuditCategory.AUTHENTICATED.toString(),
+                    AuditCategory.GRANTED_PRIVILEGES.toString());
+
+    private final boolean restAuditingEnabled;
+    private final boolean transportAuditingEnabled;
+    private final boolean resolveBulkRequests;
+    private final boolean logRequestBody;
+    private final boolean resolveIndices;
+    private final boolean excludeSensitiveHeaders;
+    private final List<String> ignoredAuditUsers;
+    private final List<String> ignoredComplianceUsersForRead;
+    private final List<String> ignoredComplianceUsersForWrite;
+    private final List<String> ignoreAuditRequests;
+    private final EnumSet<AuditCategory> disabledRestCategories;
+    private final EnumSet<AuditCategory> disabledTransportCategories;
+    private final String opendistrosecurityIndex;
+
+    public AuditConfig(final boolean restAuditingEnabled,
```

Comment:
I'd recommend using Builder pattern here as `AuditConfig` constructor takes a large number of arguments with the same type.

Reply:
I did think about this but I don't see any places where config params will be set independently. Maybe when we need it, we can have it

Reply:
If you don't want to introduce Builder now (I am OK with that), make the constructor `private`.